rllib/utils/tests/test_actor_manager.py

```diff
@@ -0,0 +1,152 @@
+import random
+import sys
+import time
+import unittest
+
+import ray
+from ray.experimental.state.api import list_actors
+from ray.rllib.utils.actor_manager import FaultTolerantActorManager
+
+
+@ray.remote(max_restarts=-1)
+class Actor:
+    def __init__(self, i):
+        random.seed(i)
+        self.count = 0
+
+    def call(self):
+        self.count += 1
+        r = random.random()
+
+        # 10% chance of crashing.
+        if r < 0.1:
+            sys.exit(1)
+        # Another 10% chance of throwing errors.
+        elif r < 0.2:
+            raise AttributeError("sorry")
+
+        # Otherwise, return good result.
+        return self.count
+
+    def apply(self, func):
+        return func(self)
+
+
+def wait_for_restore():
+    while True:
+        states = [
+            a["state"] == "ALIVE" or a["state"] == "DEAD"
+            for a in list_actors(filters=[("class_name", "=", "Actor")])
+        ]
+        print("waiting ... ", states)
+        if all(states):
+            break
+        # Otherwise, wait a bit.
+        time.sleep(0.5)
+
+
+class TestActorManager(unittest.TestCase):
+    @classmethod
+    def setUpClass(cls) -> None:
+        ray.init()
+
+    @classmethod
+    def tearDownClass(cls) -> None:
+        ray.shutdown()
+
+    def test_sync_call_healthy_only(self):
+        actors = [Actor.remote(i) for i in range(4)]
+        manager = FaultTolerantActorManager(actors=actors)
+
+        results = []
+        for _ in range(10):
+            results.extend(manager.foreach_actor(lambda w: w.call(), healthy_only=True))
+            # Wait for actors to recover.
+            wait_for_restore()
+
+        # Notice that since we only fire calls against healthy actors,
+        # we wouldn't aware that the actors have been recovered.
+        # So once an actor is taken out of the lineup (10% chance),
+        # it will not go back in, and we should have few results here.
+        # Basically takes us 24 calls to kill all the actors.
+        self.assertEqual(len(results), 24)
```

Comment:
wait, I didn't understand where this 24 is coming from

Reply:
doesn't this depend on seed?

Reply:
yeah, we set the seed in actor.__init__()

Reply:
would this not also depend on the machine / platform that runs these tests? how is the number of calls gonna be deterministic given the seed?